pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -790,6 +793,9 @@ cdef class StringHashTable(HashTable):
             else:
                 # if ignore_na is False, we also stringify NaN/None/etc.
                 v = get_c_string(<str>val)
+                if v == NULL:
```

Comment:
Can you place this in `get_c_string_buf_and_size` instead? Should keep error handling tightly coupled with where it would actually occur

Reply:
At the `get_c_string()` level the val is cast to `<str>` which I cannot give to `repr()` anymore. I tried and it crashed on me (segfault).

Reply:
Ah OK. I see your point here as well